src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -692,6 +692,111 @@ XBUtilities::str_to_reset_obj(const std::string& str)
   throw xrt_core::error(str + " is invalid. Please specify a valid reset type");
 }
 
+std::string
+XBUtilities::string_to_UUID(std::string str)
+{
+  //make sure that a UUID is passed in
+  assert(str.length() == 32);
+  std::string uuid = "";
+  //positions to insert hyphens
+  //before: 00000000000000000000000000000000
+  std::vector<int> pos = {8, 4, 4, 4};
+  //before: 00000000-0000-0000-0000-000000000000
+
+  for(auto const p : pos) {
+    std::string token = str.substr(0, p);
+    boost::to_upper(token);
+    uuid.append(token + "-");
+    str.erase(0, p);
+  }
+  boost::to_upper(str);
+  uuid.append(str);
+
+  return uuid;
+}
+
+static const std::map<uint64_t, std::string> oemid_map = {
+  {0x10da, "Xilinx"},
+  {0x02a2, "Dell"},
+  {0x12a1, "IBM"},
+  {0xb85c, "HP"},
+  {0x2a7c, "Super Micro"},
+  {0x4a66, "Lenovo"},
+  {0xbd80, "Inspur"},
+  {0x12eb, "Amazon"},
+  {0x2b79, "Google"}
+};
+
+std::string 
+XBUtilities::parse_oem_id(const std::string& oemid)
+{
+  uint64_t oem_id_val = 0;
+  std::stringstream ss;
+
+  try {
+    oem_id_val = std::stoul(oemid, nullptr, 16);
+  } catch (const std::exception&) {
+    //failed to parse oemid to hex value, ignore erros and print original value
+  }
+
+  auto oemstr = oemid_map.find(oem_id_val);
+  return oemstr != oemid_map.end() ? oemstr->second : "N/A";
+}
+
+static const std::map<std::string, std::string> clock_map = {
+  {"DATA_CLK", "Data"},
+  {"KERNEL_CLK", "Kernel"},
+  {"SYSTEM_CLK", "System"},
+};
+
+std::string 
+XBUtilities::parse_clock_id(const std::string& id)
+{
+  auto clock_str = clock_map.find(id);
+  if (clock_str != clock_map.end())
+    return clock_str->second;
+  
+  throw xrt_core::error(std::errc::invalid_argument);
+}
+
+uint64_t 
+XBUtilities::string_to_bytes(const std::string &str)
+{
+  uint64_t size = 0;
+  if(str.empty())
+    throw xrt_core::error(std::errc::invalid_argument);
+
+  uint64_t unit_bytes = 0;
+  std::string units;
+  try {
+    for(const auto& x : str) {
+      if(std::isdigit(x))
+        size = size*10 + (x - '0');
+      else if(std::isalpha(x))
+        units += x;
+    }
+    size = std::stoll(str.substr(0, str.length()-1));
```

Comment:
Doesn't this fail by given the wrong result if there are no units.  For example: "1234" would be evaluated as "123".